app/models/contact.rb

```diff
@@ -98,4 +98,16 @@ def self.contact_type_for(input_name)
   def phone_for_display
     phone.presence || cell_phone
   end
+
+  def client_opportunity_matches
+    @client_opportunity_matches ||= client_opportunity_match_contacts.joins(:match)
+  end
+
+  def open_match_count
+    client_opportunity_matches.merge(ClientOpportunityMatch.open).count
+  end
+
+  def closed_match_count
+    client_opportunity_matches.merge(ClientOpportunityMatch.closed).count
+  end
```

Comment:
These are going to introduce N+1, so we should see if we can batch calculate.  I think something in the controller is probably easiest.  This should happen after the pagination.
```
@open_matches = Contact.where(id: @contacts.map(&:id)).joins(:matches).merge(ClientOpportunityMatch.open).group(:id).count
@closed_matches = Contact.where(id: @contacts.map(&:id)).joins(:matches).merge(ClientOpportunityMatch.closed).group(:id).count
```
Those would then be keyed on contact ID.

Reply:
thanks, updated to use this